mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala

```diff
@@ -17,16 +17,27 @@
 
 package org.apache.spark.ml.source.libsvm
 
+import java.io.{CharArrayWriter, IOException}
+
+import com.fasterxml.jackson.core.JsonFactory
 import com.google.common.base.Objects
+import org.apache.hadoop.fs.{Path, FileStatus}
```

Comment:
I don't know we are checking the ordering of imports now. `F` should come before `P`. You can use Scala Import Organizer with IntelliJ to organize imports quickly.